pubsub/rabbitmq/metadata.go

```diff
@@ -184,3 +217,25 @@ func (m *metadata) formatQueueDeclareArgs(origin amqp.Table) amqp.Table {
 func exchangeKindValid(kind string) bool {
 	return kind == amqp.ExchangeFanout || kind == amqp.ExchangeTopic || kind == amqp.ExchangeDirect || kind == amqp.ExchangeHeaders
 }
+
+func (m *metadata) connectionURI() string {
+	if m.connectionString != "" {
+		return m.connectionString
+	}
+
+	amqpAuthority := strings.Builder{}
+
+	if m.username != "" {
+		amqpAuthority.WriteString(m.username)
+
+		if m.password != "" {
+			amqpAuthority.WriteString(":" + m.password)
+		}
+
+		amqpAuthority.WriteString("@")
+	}
+
+	amqpAuthority.WriteString(m.hostname)
+
+	return fmt.Sprintf("%s://%s", m.protocol, amqpAuthority.String())
```

Comment:
I also wonder if we should use `url` to build this, which would also take care of correctly escaping sequences when needed? WDYT?

Something like:

```go
		u := url.URL{}
		u.Scheme = m.protocol
		u.Host = m.hostname
		if m.username != "" {
			u.User = ...
		}
		return u.String()
```